agentlocal/agent_local.go

```diff
@@ -53,6 +65,17 @@ func (als *AgentLocalServer) getMetadata() agentpb.AgentServerMetadata {
 	return *als.serverMetadata
 }
 
+func (als *AgentLocalServer) getAgentsList() (res []*agentlocalpb.AgentInfo, err error) {
+	als.rw.RLock()
+	defer als.rw.RUnlock()
+
+	if als.ag == nil {
+		panic("agentsGetter is nil. Use SetAgentsGetter(ag agentsGetter) to set it.")
```

Comment:
Can we pass that parameter to constructor? In general, half-constructed objects are bad.

Reply:
> Can we pass that parameter to the constructor?  In general, half-constructed objects are bad.
Maybe, I don't know, but in general, there is nothing wrong with it. It's simple https://en.wikipedia.org/wiki/Dependency_injection#Setter_injection